### ISSUE TITLE
[Question] How to stop NeMo from printing the result?

### ISSUE THREAD

**Issue by soheiltehranipour** [NVIDIA/NeMo#1274]
**Describe your question**

After start training, it keeps printing the loss , reference and decoded sentences and eventually after 30 minutes or so, the browser crashes because of the number of lines printed as the cell's output in google colab.
How can I stop it from printing the result to solve this issue?

**Environment overview:**
-Environment location: Google Colab
-Method of NeMo install: !python -m pip install git+https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]

**Environment details**
-OS version : Ubuntu 18.04
-PyTorch version : 1.6.0+cu101
-Python version : Python 3.6.9
-GPU model: Tesla T4

**Comment by titu1994** [comment]
You can change `trainer.row_low_interval` to a high value (say 50). You can also set scrolling enabled on the cell in Jupyter or Colab so that it doesn't overflow with number of lines.

**Comment by soheiltehranipour** [comment]
@titu1994 I could not do that because there is no option for trainer about row low interval. Am I doing something wrong?
I even read whole the PyTorch lightning document and unfortunately there was not anything like "row_low_interval" for trainer.

Thank you in advance.

**Comment by titu1994** [comment]
It's my mistake. It's trainer row_log_interval in the config.

**Comment by usamaehsan** [comment]
@titu1994 i have changed trainer.row_low_interval to 100 but still its printing every result:

[NeMo I 2020-10-21 16:07:21 wer:160] reference:سیکنڈ چیپٹر مدافعانہ جنگ گزشتہ بحث سے یہ
[NeMo I 2020-10-21 16:07:21 wer:161] decoded  :سیکن چیپٹر مدافیانہ جنگ گذشتہ بحت سے یہ
[NeMo I 2020-10-21 16:07:22 wer:159]

**Comment by titu1994** [comment]
Are you training from scratch? Changing the config to have row_log_interval > 1 makes it print every N steps where N < number of batches in one epoch. If you are using a notebook, I assume you are fine tuning? In that case you also need to attach the trainer to the model - via model.set_trainer(trainer).

**Comment by titu1994** [comment]
Closing this issue for lack of update, if this issue is not resolved, please reopen.

**Comment by rbracco** [comment]
I am also trying to stop the result from printing so frequently but can't seem to get it working. Here's what I've tried. 

```
trainer = pl.Trainer(gpus=1, max_epochs=10)
trainer.row_log_interval = 100
trainer.fit(quartznet)
```

and 

```
# load a previously saved model from checkpoint
trainer = pl.Trainer(gpus=1, max_epochs=10)
trainer.row_log_interval = 100
checkpoint_name = 'quartznet12_67.nemo'
quartznet_empty_model = nemo_asr.models.EncDecCTCModel(cfg=DictConfig(params['model']), trainer=trainer)
quartznet = quartznet_empty_model.restore_from(f'{path_saved_models}/{checkpoint_name}')
```

I've added `row_log_interval` to my config but I'm not sure how to instantiate a new lightning Trainer from the config. Can you give a concrete example? Thank you!

**Comment by titu1994** [comment]
If you are using Nemo 1.0.0b2, row log interval is deprecated by Pytorch Lightning. You'll need to use the new `log_every_n_steps`.

The value just be passed to the constructor of the Trainer. 

Note - Restore from is a class method, and will remove the trainer from the model! You need to reassign the trainer using quartznet set_trainer().

**Comment by rbracco** [comment]
Awesome thanks again for all your help. I was using Nemo 1.0.0b1 by mistake, probably from following an old version of a tutorial. Here's an example of what worked for me in the end:

```
trainer = pl.Trainer(gpus=1, max_epochs=10, log_every_n_steps=100)
quartznet.set_trainer(trainer)
trainer.fit(quartznet)
```
